packages/gatsby-remark-embed-snippet/src/index.js

```diff
@@ -108,6 +127,19 @@ module.exports = (
 
           return returnValue
         })
+        .filter((line, index) => {
+          // (Hide and highlight line numbers are 1-based).
+          const isHiddenLine = hiddenLines.includes(index + 1)
+          const isHighlightedLine = highlightLines.includes(index + 1)
+          // check if hidden and highlighted lines overlap
+          const hasIntersection = isHiddenLine && isHighlightedLine
+
+          if (hasIntersection) {
+            throw Error(`There can be no intersection between hidden lines and highlighted lines - "${path}"`)
```

Comment:
What about `The number of range cannot be intersected between hidden lines and highlighted lines. Check the file: ${path}.`

Reply:
Done!